node-graph/gcore/src/raster/adjustments.rs

```diff
@@ -924,6 +924,26 @@ fn generate_curves<_Channel: Channel + super::Linear>(_primary: (), curve: Curve
 	ValueMapperNode::new(lut)
 }
 
+#[derive(Debug, Clone)]
+pub struct ColorFillNode<C> {
+	color: C,
+}
+
+#[node_macro::node_fn(ColorFillNode)]
+pub fn color_fill_node(image_frame: ImageFrame<Color>, color: Color) -> ImageFrame<Color> {
+	let target_width = (image_frame.transform.transform_vector2((1., 0.).into()).length() as usize).min(image_frame.image.width as usize);
+	let target_height = (image_frame.transform.transform_vector2((0., 1.).into()).length() as usize).min(image_frame.image.height as usize);
+
+	for mut pixel in &image_frame.image.data {
+		pixel = &color;
```

Comment:
If you want to preserve the opacity, you could change this code to be something like
```rs
pixel.set_red(color.red());
pixel.set_green(color.green());
pixel.set_blue(color.blue());
```
After importing the correct traits.